frontend/src/metabase/modes/components/drill/ObjectDetailDrill.jsx

```diff
@@ -74,16 +49,11 @@ function getPKAction({ question, column, objectId, isDashboard }) {
 
 function getFKAction({ question, column, objectId }) {
   const actionObject = getBaseActionObject();
-  const targetField = getFKTargetField(question, column);
-  if (!targetField) {
+  const fkField = question.metadata().field(column.id);
+  if (!fkField?.target) {
     return;
   }
-  const [actionKey, action] = getActionForFKColumn({
-    field: targetField,
-    objectId,
-  });
-
-  actionObject[actionKey] = action;
+  actionObject.question = () => question.drillPK(fkField.target, objectId);
```

Comment:
We used to use this approach before #20101 (see [code](https://github.com/metabase/metabase/blob/8f119d6457f70827a40b105a60218d855a030969/frontend/src/metabase/modes/components/drill/ObjectDetailDrill.jsx#L31)), so I suppose it would match the previous behaviour better